### PR TITLE
feat: Update RN troubleshooting docs for promise polyfill fix

### PR DESCRIPTION
**Pull request by jennmueng**
Documents auto promise patching added in https://github.com/getsentry/sentry-react-native/pull/1984 and demonstrates how to disable the patching or how to use the patching with other polyfilling libraries.

### CODE REVIEW

File: src/platforms/react-native/troubleshooting.mdx
Comment (bruno-garcia) [inline src/platforms/react-native/troubleshooting.mdx:43]: Worth adding this option to the Configuration page too.

**Review by bruno-garcia — APPROVED** [review]
Nice! We can merge this after the holidays when we merge the PR in the RN repo and make a release there

File: src/platforms/react-native/troubleshooting.mdx
Comment (imatwawana) [inline src/platforms/react-native/troubleshooting.mdx:23]:
```suggestion
If you use a polyfilling library that patches the global `Promise` instance, you'll need to make sure you run the polyfill **after** `Sentry.init` is called.
```

File: src/platforms/react-native/troubleshooting.mdx
Comment (imatwawana) [inline src/platforms/react-native/troubleshooting.mdx:39]:
```suggestion
Your linter might throw some errors here, but this step is necessary.
```

File: src/platforms/react-native/troubleshooting.mdx
Comment (imatwawana) [inline src/platforms/react-native/troubleshooting.mdx:43]:
```suggestion
You can disable the global promise patching by passing `patchGlobalPromise: false` in either `Sentry.init` or the `ReactNativeErrorHandlers` integration. Note that if you disable our auto patching, to ensure that unhandled rejections are still caught, you will need to [manually force a package resolution](#manually-forcing-a-package-resolution).
```

File: src/platforms/react-native/troubleshooting.mdx
Comment (imatwawana) [inline src/platforms/react-native/troubleshooting.mdx:47]:
```suggestion
You don't need to perform the steps below if you don't disable [auto patching](#auto-patching-default-behavior). You'll need to ensure that the version of `promise` that you use matches exactly with the version that React Native uses.
```

File: src/platforms/react-native/troubleshooting.mdx
Comment (imatwawana) [inline src/platforms/react-native/troubleshooting.mdx:76]:
```suggestion
3. If the fix is successful, our SDK will no longer display the above warning and will indicate that promise rejections will be caught.
```

**Review by imatwawana — APPROVED** [review]
Some minor wording changes

**Pull request merged**
